xds/internal/xdsclient/xdslbregistry/xdslbregistry_test.go

```diff
@@ -317,6 +336,10 @@ func (s) TestConvertToServiceConfigSuccess(t *testing.T) {
 				defer func(old bool) { envconfig.XDSRingHash = old }(envconfig.XDSRingHash)
 				envconfig.XDSRingHash = false
 			}
+			if test.lrDisabled {
+				defer func(old bool) { envconfig.LeastRequestLB = old }(envconfig.LeastRequestLB)
+				envconfig.LeastRequestLB = false
+			}
```

Comment:
Default value of `envconfig.LeastRequestLB` is already `false`. You are setting it to true for the duration of the test at the top, and then setting it to false here again.

Why not have a `lrEnabled` field which is set to true only for tests that require it, and here, you can check for `if test.lrEnabled { // set the env var to true }`

Reply:
Done :). Good suggestion.